site/content/how-to/monitoring/troubleshooting.md

```diff
@@ -65,54 +65,297 @@ LAST SEEN   TYPE      REASON              OBJECT
 ```
 
+##### Get shell access to containers
+
+Getting shell access to containers allows developers and operators to view the environment of a running container, see its logs or diagnose any problems. To get shell access to the NGINX container, use `kubectl exec`:
+
+```shell
+kubectl exec -it [-n namespace] <ngf-pod-name> -c nginx /bin/sh
+```
+
 ##### Logs
 
 Logs from the NGINX Gateway Fabric control plane and data plane can contain information that isn't available to status or events. These can include errors in processing or passing traffic.
 
-To see logs for the control plane container:
+1. To see logs for the control plane container:
 
 ```shell
-kubectl -n nginx-gateway logs <ngf-pod-name> -c nginx-gateway
+kubectl [-n namespace] logs <ngf-pod-name> -c nginx-gateway
 ```
 
 To see logs for the data plane container:
 
 ```shell
-kubectl -n nginx-gateway logs <ngf-pod-name> -c nginx
+kubectl [-n namespace] logs <ngf-pod-name> -c nginx
 ```
 
+1. To only see error logs for control plane and data plane containers:
+
+For _nginx-gateway_ container, you can `grep` for the word `error` or change the log level to `error` by following steps in [Modify logging levels](#modify-logging-levels).
+
+```shell
+kubectl [-n namespace] logs <ngf-pod-name> -c nginx-gateway | grep error
+```
+
+For example, an error message when telemetry is not enabled for NGINX Plus installations:
+
+```text
+kubectl logs -n nginx-gateway nginx-gateway-nginx-gateway-fabric-77f8746996-j6z6v | grep error
+Defaulted container "nginx-gateway" out of: nginx-gateway, nginx
+{"level":"error","ts":"2024-06-13T18:22:16Z","logger":"usageReporter","msg":"Usage reporting must be enabled when using NGINX Plus; redeploy with usage reporting enabled","error":"usage reporting not enabled","stacktrace":"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static.createUsageWarningJob.func1\n\tgithub.com/nginxinc/nginx-gateway-fabric/internal/mode/static/manager.go:616\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:259\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:226\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:227\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:204\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\tk8s.io/apimachinery@v0.30.1/pkg/util/wait/backoff.go:259\ngithub.com/nginxinc/nginx-gateway-fabric/internal/framework/runnables.(*CronJob).Start\n\tgithub.com/nginxinc/nginx-gateway-fabric/internal/framework/runnables/cronjob.go:53\nsigs.k8s.io/controller-runtime/pkg/manager.(*runnableGroup).reconcile.func1\n\tsigs.k8s.io/controller-runtime@v0.18.4/pkg/manager/runnable_group.go:226"}
+```
+
+For _nginx_ container:
+
+```shell
+kubectl [-n namespace] logs <ngf-pod-name> -c nginx-gateway | grep emerg
```

Comment:
```suggestion
kubectl [-n namespace] logs <ngf-pod-name> -c nginx | grep emerg
```

Reply:
Let's make a note of all the error log levels a user can search for: https://nginx.org/en/docs/ngx_core_module.html#error_log.

They are all logged in brackets, e.g.  `[error]`